### PR TITLE
added some types for node properties

### PR DESCRIPTION
**Pull request by smoka7**

### CODE REVIEW

File: types.go [hunk 1]
```diff
@@ -132,21 +178,18 @@ func (n *Node) FocusedNode() *Node {
...
+	}
+	for _, node := range n.Nodes {
+		if n := node.TraverseNodes(predicate); n != nil {
+			return n
 		}
-
-		if predicate(n) {
+	}
+	for _, node := range n.FloatingNodes {
+		if n := node.TraverseNodes(predicate); n != nil {
```
Comment (joshuarubin) [inline types.go]: Why the change from BFS to DFS?

Comment (smoka7) [reply types.go]: Using DFS in here doesn't require the slice operations so it's much faster than BFS.

Comment (joshuarubin) [reply types.go]: let's just keep the changes to the types

**Review by joshuarubin — APPROVED** [review]
